src/models/propulsion/FGBrushLessDCMotor.cpp

```diff
@@ -0,0 +1,236 @@
+/*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+ Module:       FGBrushLessDCMotor.cpp
+ Autor         Paolo Becchi
+ 1st release   1/1/2022
+ Purpose:      This module models an BLDC electric motor
+
+  ------------- Copyright (C) 2022  Paolo Becchi (pbecchi@aerobusinees.it) -------------
+
+  This program is free software; you can redistribute it and/or modify it under
+  the terms of the GNU Lesser General Public License as published by the Free
+  Software Foundation; either version 2 of the License, or (at your option) any
+  later version.
+
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+  FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+  details.
+
+  You should have received a copy of the GNU Lesser General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc., 59
+  Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+
+  Further information about the GNU Lesser General Public License can also be
+  found on the world wide web at http://www.gnu.org
+
+FUNCTIONAL DESCRIPTION
+--------------------------------------------------------------------------------
+Following code represent a new BrushLess DC motor to be used as alternative 
+to basic electric motor. 
+BLDC motor code is based on basic "3 constant motor equations"
+It require 3 basic physical motor properties:
+Kv speed motor constant
+Rm internal coil resistance
+I0 no load current
+
+HISTORY
+--------------------------------------------------------------------------------
+1/01/2022    Created
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+INCLUDES
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+#include <iostream>
+#include <sstream>
+#include <math.h>
+
+#include "FGFDMExec.h"
+#include "FGBrushLessDCMotor.h"
+#include "FGPropeller.h"
+#include "input_output/FGXMLElement.h"
+
+using namespace std;
+
+namespace JSBSim {
+
+/*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+CLASS IMPLEMENTATION
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+FGBrushLessDCMotor::FGBrushLessDCMotor(FGFDMExec* exec, Element* el, int engine_number, struct FGEngine::Inputs& input)
+  : FGEngine(engine_number, input)
+{
+  Load(exec, el);
+
+  Type = etElectric;
+  PowerWatts = 745.7;
+  hptowatts = 745.7;
+
+
+  if (el->FindElement("maxvolts"))
+    MaxVolts = el->FindElementValueAsNumber("maxvolts");
+
+  if (el->FindElement("velocityconstant"))
+    VelocityConstant = el->FindElementValueAsNumber("velocityconstant");
```

Comment:
```suggestion
  if (el->FindElement("velocityconstant"))
    VelocityConstant = el->FindElementValueAsNumber("velocityconstant");
  else {
    cerr << el->ReadFrom()
         << "<velocityconstant> is a mandatory parameter" << endl;
    throw JSBBaseException("Missing parameter");
  }
```
There is no default value for `VelocityConstant` so you must throw if the user does not specify it.